CMakeLists.txt

```diff
@@ -14,11 +14,32 @@ find_package(TinyXML REQUIRED)
 include_directories(include ${TinyXML_INCLUDE_DIRS} ${catkin_INCLUDE_DIRS} ${console_bridge_INCLUDE_DIRS} ${urdfdom_headers_INCLUDE_DIRS})
 link_directories(${catkin_LIBRARY_DIRS})
 
+
+### Maintain compatibility to old urdfdom_headers (version < 4.0)
+# check for existence of type urdf::LinkSharedPtr
+include(CheckCXXSourceCompiles)
+check_cxx_source_compiles("
+#include <urdf_model/model.h>
+
+int main(int argc, char** argv)
+{
+  urdf::LinkSharedPtr l;
+  return 0;
+}
+" HAVE_URDFDOM_4)
```

Comment:
Will this break if any single character changes in the upstream file? Seems really unstable!